cmd/kube-apiserver/app/server.go

```diff
@@ -689,6 +690,9 @@ func Complete(s *options.ServerRunOptions) (completedServerRunOptions, error) {
 				s.Authentication.ServiceAccounts.MaxExpiration > upBound {
 				return options, fmt.Errorf("the serviceaccount max expiration must be between 1 hour to 2^32 seconds")
 			}
+			if s.Authentication.ServiceAccounts.ExtendExpiration && s.Authentication.ServiceAccounts.MaxExpiration < serviceaccount.ExpirationExtensionSeconds*time.Second {
```

Comment:
only warn if `s.Authentication.ServiceAccounts.MaxExpiration > 0`

Reply:
also, we do need to ensure that service-account-max-token-expiration is not shorter than WarnOnlyBoundTokenExpirationSeconds, otherwise the max expiration shortens the requested token time and breaks the magic expiration detection in admission

Reply:
The warning is already in the MaxExpiration != 0 branch, and there is a check for upper/lower bound before. so we don't need to check MaxExpiration > 0 again.
Added the warning if MaxExpiration < WarnOnlyBoundTokenExpirationSeconds. Otherwise the safe transition logic cannot be triggered. Use warning instead of error in case it breaks some edge case clients who set really short max expiration.